crates/runtime/benches/bench_search/setup.rs

```diff
@@ -93,6 +103,103 @@ pub(crate) async fn load_search_queries(rt: &Runtime) -> Result<Vec<String>, Str
     Ok(queries)
 }
 
+pub(crate) async fn load_query_relevance_data(rt: &Runtime) -> Result<QueryRelevance, String> {
+    let test_queries = rt
+        .datafusion()
+        .query_builder(r#"SELECT "query-id", "corpus-id", score FROM test_score"#)
+        .build()
+        .run()
+        .await
+        .map_err(|e| format!("Failed to retrieve test scores: {e}"))?;
+
+    let records = test_queries
+        .data
+        .try_collect::<Vec<RecordBatch>>()
+        .await
+        .map_err(|e| format!("Failed to retrieve test scores: {e}"))?;
+
+    let qrels = extract_query_relevance_from_batches(&records)?;
+
+    tracing::info!(
+        "Loaded benchmark query relevance data for {num_rows} queries",
+        num_rows = qrels.len()
+    );
+
+    Ok(qrels)
+}
+
+fn extract_queries_from_batches(records: &[RecordBatch]) -> Result<Vec<Query>, String> {
+    let queries = records
+        .iter()
+        .map(|batch| {
+            let id_column = batch
+                .column(0)
```

Comment:
nit: could use `.column_by_name("id")` to make it more robust

Reply:
Updated - thank you!!!